### DISCUSSION TITLE
embed and execute CMD command in c# app silently

### DISCUSSION THREAD

**Discussion by Omar7saeed** [Tyrrrz/CliWrap#115] [Q&A] [answered]
hello
I never post issues or discussions here
But I found that you always reply on the same day
This attracted me

I'm a beginner.. just started learning 
I want to embed & execute CMD command in net framework 4.6.1

i have tried many things for hours 
lastest was

`      async Task test()
            {

                var result = await Cli.Wrap("cmd")
                .WithArguments("SET PS2CMD_ARGS=%* & PowerShell.exe -NoProfile -EncodedCommand VwByAGkAdAlAC0A..ETC")
                .ExecuteBufferedAsync();

                 var stdOut = result.StandardOutput;
            }`

no compiling errors 
but the cmd command didn't run 

dear Alexey can u tell me how I can do this correctly

**Comment by Tyrrrz** [comment, marked as answer]
Why not just launch powershell directly? I.e.:

```csharp
var result = await Cli.Wrap("powershell")
            .WithArguments(new[] {"-NoProfile", "-EncodedCommand", "VwByAGkAdAlAC0A..ETC"})
            .ExecuteBufferedAsync();
```

**Comment by Omar7saeed** [comment]
thanks, brother 
yea everything is fine now 💯 

notice to anyone interesting : I replaced 

async Task test()
{
        }

with 

`static async System.Threading.Tasks.Task Main()
        {
                  }`


I want to take your advice on something if u don't mind .. sent u on Twitter "xeros"
thanks again and 

**Best regards**

**Reply by Tyrrrz** [reply]
You're welcome. Sorry, I don't reply to code-related questions on Twitter usually. If they are related to one of my projects, please ask in the corresponding discussion section instead.
